File: backend/src/main/java/com/example/backend/security/Role.java
package com.example.backend.security;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

public enum Role {
    PATIENT,
    RECEPTIONIST,
    DOCTOR,
    ADMIN;

    public GrantedAuthority authority() {
        return new SimpleGrantedAuthority("ROLE_" + name());
    }
}

File: frontend/lib/auth/role.dart
/// Mirrors the backend `Role` enum; shapes the UI only, the backend still authorizes.
enum Role {
  patient('PATIENT'),
  receptionist('RECEPTIONIST'),
  doctor('DOCTOR'),
  admin('ADMIN');

  const Role(this.wireName);

  /// The value the backend sends, matching `Role.name()` on the server.
  final String wireName;

  /// Returns null for an unrecognised role, which callers reject rather than guess at.
  static Role? tryParse(String? value) {
    for (final role in Role.values) {
      if (role.wireName == value) {
        return role;
      }
    }
    return null;
  }

  bool get isClinicStaff =>
      this == Role.doctor || this == Role.receptionist || this == Role.admin;
}
